### ISSUE TITLE
Use after free if pk_client_copy_downloaded_file() fails

### ISSUE THREAD

**Issue by hughsie** [hughsie/PackageKit#312]
`pk_client_copy_downloaded_file()` calls `pk_client_state_finish()` if the file already exists. This could cause multiple invocations of `pk_client_state_finish()` to happen. We just need to remove the async copying and just do it serially.

**Comment by ximion** [comment]
Is this actually an issue? `pk_client_state_finish` tests for `if (state->res == NULL)` and simply does nothing if invoked multiple times. Or is this whole thing threaded? In this case, putting `pk_client_state_finish` behind a mutex may also do the trick.

**Comment by hughsie** [comment]
> Is this actually an issue

Yes, but thankfully not much uses `DOWNLOAD_PACKAGES` -- and I'm wondering if we should remove this dodgy API completely.
